onnxscript/testing/__init__.py

```diff
@@ -448,6 +449,14 @@ def assert_onnx_proto_equal(
                         error_message = f"Field {field} index {i} in sequence not equal. type(a_value_i): {type(a_value_i)}, type(b_value_i): {type(b_value_i)}, a_value_i: {a_value_i}, b_value_i: {b_value_i}"
                         raise AssertionError(error_message) from e
                 elif a_value_i != b_value_i:
+                    if (
+                        isinstance(a_value_i, float)
+                        and isinstance(b_value_i, float)
+                        and math.isnan(a_value_i)
+                        and math.isnan(b_value_i)
+                    ):
+                        # Consider NaNs equal
```

Comment:
Does this mean ONNX IR could get different results/values from the original one (protobuf?)

Reply:
There was a test cases in ONNX that contained NaN. Since NaN != NaN when directly compared, I created this to handle it. They should be considered equal for our purposes.